auth/caios-login/index.js

```diff
@@ -0,0 +1,126 @@
+const core = require("@actions/core");
+const exec = require("@actions/exec");
+
+async function run() {
+  try {
+    // Get inputs
+    const region = core.getInput("region", { required: true });
+    const orgId = core.getInput("org_id", { required: true });
+    const profile = core.getInput("profile") || "";
+    const audience = core.getInput("audience") || "https://coreweave.com/iam";
+    const endpoint =
+      core.getInput("endpoint") ||
+      "https://api.coreweave.com/v1/cwobject/temporary-credentials/oidc";
+    const s3Endpoint = core.getInput("s3_endpoint") || "https://cwobject.com";
+
+    // Step 1: Get ID Token
+    core.info("Getting ID token...");
+    const idToken = await core.getIDToken(audience);
+
+    // Step 2: Get credentials from CAIOS
+    core.info("Fetching credentials from CAIOS...");
+    const response = await fetch(`${endpoint}/${orgId}`, {
+      headers: {
+        Authorization: idToken,
+      },
+    });
+
+    if (!response.ok) {
+      const errorText = await response.text();
+      throw new Error(`HTTP ${response.status}: ${errorText}`);
+    }
+
+    const credentials = await response.json();
+
+    // Step 3: Export credentials to environment
+    core.info(`Setting AWS credentials for ${credentials.AccessKeyID}...`);
+
+    // Only export to environment if using default profile
+    if (!profile || profile === "default") {
+      core.exportVariable("AWS_ACCESS_KEY_ID", credentials.AccessKeyID);
+
+      // Mask the secret key
+      core.setSecret(credentials.SecretAccessKey);
+      core.exportVariable("AWS_SECRET_ACCESS_KEY", credentials.SecretAccessKey);
+
+      // If session token exists, export it too
+      if (credentials.Token) {
+        core.setSecret(credentials.Token);
+        core.exportVariable("AWS_SESSION_TOKEN", credentials.Token);
+      }
+    } else {
+      // Still mask the secrets even when using profiles
+      core.setSecret(credentials.SecretAccessKey);
+      if (credentials.Token) {
+        core.setSecret(credentials.Token);
+      }
+    }
+
+    // Set output for other steps to use
+    core.setOutput("access_key_id", credentials.AccessKeyID);
+
+    // Step 4: Configure AWS CLI
+    const profileArgs =
+      profile && profile !== "default" ? ["--profile", profile] : [];
+    core.info(
+      `Configuring AWS CLI${profileArgs.length ? ` for profile: ${profile}` : ""}...`,
+    );
+
+    // Set AWS configuration
+    await exec.exec("aws", [
```

Comment:
Before we execute these shell cmd please have logic to ensure all prerequisites are met. ie) required inputs, validation, `aws` executable exists, etc... Please put it in a helper method that is called at the beginning of the action :)